src/compat/posix/fnmatch/fnmatch.c

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file
+ * @brief fnmatch_function
+ *
+ * @date 20.12.18
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <fnmatch.h>
+
+static int bracket_check(const char *pattern) {
+	while (*pattern != ']') {
+		if (*pattern == '/') {
+			return 0;
+		}
+		pattern++;
+	}
+	return 1;
+}
+
+static int fnmatch_string(const char *pattern, const char *string, int flags) {
+	int flag = 0;
+	int denial_flag = 0;
+	while (*pattern != '\0') {
+		flag = 0;
+		denial_flag = 0;
+
+		if ((*string == '\0') && (*pattern != '*')) {
+			return FNM_NOMATCH;
+		}
+		switch (*pattern) {
+
+			case '?':
+				break;  
+
+			case '*':
+				pattern++;
+				if (*pattern == '\0') {
+					return 0;
+				}
+				while (*string != '\0') {
+					if (!fnmatch_string(pattern, string, flags)) {
+						return 0;
+					}
+					string++;
+				}
+				return FNM_NOMATCH;
+
+			case '/':
+				if (*pattern != *string) {
+					return FNM_NOMATCH;
+				}
+				if (flags == FNM_PATHNAME) {
+					return 0;
+				}
+				break;		
+
+			case '[':
+				if (flags == FNM_PATHNAME) {
+					if (!bracket_check(pattern)) {
+						if (*pattern != *string) {
+							return FNM_NOMATCH;
+						}
+						pattern++;
+						string++;
+						return fnmatch_string(pattern, string, flags);
```

Comment:
You can replace these 3 lines with this:
`return fnmatch_string(pattern + 1, string +1, flags); `

I think it looks better.

Reply:
okey - changed

Reply:
if I understand correctly-the function will work with the old values and then make '+1' the pattern and the string, so it will not work correctly if i will change

Reply:
Nope, this problem raises only when you use `++`/`--`. I.e. if you call `fnmatch_string(pattern++,..)` than old value will be passed (`fnmatch_string(++pattern,...` will be correct, but it's pretty bad code style), but when you use `pattern + 1` (or any other expression without pre/post increment/decrement operator), then at first expression is calculated and then function is called with that value.

Reply:
changed